pkg/cfn/builder/api_test.go

```diff
@@ -2933,6 +2939,94 @@ var _ = Describe("CloudFormation template builder API", func() {
 
 		})
 	})
+
+	Context("NodeGroup{T3Unlimited=nil}", func() {
+		cfg, ng := newClusterConfigAndNodegroup(true)
+
+		build(cfg, "eksctl-test-t3-unlimited", ng)
+
+		roundtrip()
+
+		It("should have correct resources and attributes", func() {
+			Expect(getLaunchTemplateData(ngTemplate).CreditSpecification).To(BeNil())
+		})
+	})
+
+	Context("NodeGroup{T3Unlimited=false InstancesDistribution.InstanceTypes=t3.medium,t3a.medium}", func() {
+		cfg, ng := newClusterConfigAndNodegroup(true)
+
+		ng.InstanceType = "mixed"
+		ng.T3Unlimited = false
+		ng.InstancesDistribution = &api.NodeGroupInstancesDistribution{
+			MaxPrice:                            &maxSpotPrice,
+			InstanceTypes:                       []string{"t3.medium", "t3a.medium"},
+			OnDemandBaseCapacity:                &baseCap,
+			OnDemandPercentageAboveBaseCapacity: &percentageOnDemand,
+			SpotInstancePools:                   &pools,
+			SpotAllocationStrategy:              &spotAllocationStrategy,
+		}
+
+		build(cfg, "eksctl-test-t3-unlimited", ng)
+
+		roundtrip()
+
+		It("should have correct resources and attributes", func() {
+			Expect(getLaunchTemplateData(ngTemplate).CreditSpecification).To(BeNil())
+			Expect(getLaunchTemplateData(ngTemplate).CreditSpecification.CpuCredits).ToNot(BeNil())
+			Expect(getLaunchTemplateData(ngTemplate).CreditSpecification.CpuCredits).To(Equal("standard"))
+		})
+	})
+
+	Context("NodeGroup{T3Unlimited=true InstancesDistribution.InstanceTypes=t3.medium,t3a.medium}", func() {
+		cfg, ng := newClusterConfigAndNodegroup(true)
+
+		ng.InstanceType = "mixed"
+		ng.T3Unlimited = true
+		ng.InstancesDistribution = &api.NodeGroupInstancesDistribution{
+			MaxPrice:                            &maxSpotPrice,
+			InstanceTypes:                       []string{"t3.medium", "t3a.medium"},
+			OnDemandBaseCapacity:                &baseCap,
+			OnDemandPercentageAboveBaseCapacity: &percentageOnDemand,
+			SpotInstancePools:                   &pools,
+			SpotAllocationStrategy:              &spotAllocationStrategy,
+		}
+
+		build(cfg, "eksctl-test-t3-unlimited", ng)
+
+		roundtrip()
+
+		It("should have correct resources and attributes", func() {
+			Expect(getLaunchTemplateData(ngTemplate).CreditSpecification).ToNot(BeNil())
+			Expect(getLaunchTemplateData(ngTemplate).CreditSpecification.CpuCredits).ToNot(BeNil())
+			Expect(getLaunchTemplateData(ngTemplate).CreditSpecification.CpuCredits).To(Equal("unlimited"))
+		})
+	})
+
+	Context("NodeGroup{T3Unlimited=true InstancesDistribution.InstanceTypes=m5.large}", func() {
```

Comment:
We should prevent this combination earlier on during the validation of the yaml file so that there are no unexpected subtle changes like changing the value of a field internally.  Please take a look at `validation.go` and `validation_test.go` to see how to do this.

Reply:
Thanks, will take a look.

Reply:
Fixed in latest commit